orchagent/bufferorch.cpp

```diff
@@ -520,17 +546,292 @@ task_process_status BufferOrch::processEgressBufferProfileList(Consumer &consume
     return task_process_status::task_success;
 }
 
+/*
+Input sample "BUFFER_PORT_CABLE_LENGTH_TABLE:"
+             "port_alias" : "cable_length"
+*/
+task_process_status BufferOrch::processPortCableLenth(Consumer &consumer)
+{
+    SWSS_LOG_ENTER();
+
+    KeyOpFieldsValuesTuple tuple = consumer.m_toSync.begin()->second;
+    string key = kfvKey(tuple);
+    string op = kfvOp(tuple);
+
+    if (op != SET_COMMAND)
+    {
+        return task_process_status::task_invalid_operation;
+    }
+
+    for (auto itp : kfvFieldsValues(tuple))
+    {
+        Port port;
+        string port_name = fvField(itp);
+        string cable_length = fvValue(itp);
+
+        if (!m_portsOrch->getPort(port_name, port))
+        {
+            SWSS_LOG_ERROR("Port with alias:%s not found", port_name.c_str());
+            return task_process_status::task_invalid_entry;
+        }
+
+        m_cableLengthMap[port_name] = cable_length;
+    }
+
+    return task_process_status::task_success;
+}
+
+/*
+Input sample "PORT_CONFIG_TO_PG_PROFILE_TABLE:"
```

Comment:
why all the profile information needs to be stored in the orchagent? it could be better to have some file store all the profile/cable length mapping and pick the correct one and then configure it to the orchagent. then, the application database will only have the corresponding profile that is currently used.